web3.js/src/connection.ts

```diff
@@ -3304,6 +3304,34 @@ export class Connection {
     return res;
   }
 
+  /**
+   * Fetch transaction details for a batch of confirmed transactions.
+   * Similar to {@link getParsedTransactions} but returns a {@link TransactionResponse}.
+   */
+  async getTransactions(
+    signatures: TransactionSignature[],
+    commitment?: Finality,
+  ): Promise<(TransactionResponse | null)[]> {
+    const batch = signatures.map(signature => {
+      const args = this._buildArgsAtLeastConfirmed([signature], commitment);
+      return {
+        methodName: 'getTransaction',
+        args,
+      };
+    });
+
+    const unsafeRes = await this._rpcBatchRequest(batch);
+    const res = unsafeRes.map((unsafeRes: any) => {
```

Comment:
Somehow, the type of `res` here is `any`, despite TypeScript knowing the type of `res.result`.

<img width="533" alt="image" src="https://user-images.githubusercontent.com/13243/174849792-6ee4bff3-15dc-480e-8a70-88fb4629952c.png">

Fixed in #26099.

Reply:
I wonder if this is because both variables are named the same, and one shadows the other.

Reply:
If true, we should definitely install the `no-shadow` linter. https://eslint.org/docs/rules/no-shadow

Reply:
I figured out why this happened. We had insufficient type coverage here, so you missed this error. https://github.com/solana-labs/solana/pull/26102